public/app/features/explore/QueryLibrary/types.ts

```diff
@@ -0,0 +1,11 @@
+import { ComponentType } from 'react';
+
+import { DataQuery } from '@grafana/schema';
+
+export type QueryActionButtonProps = {
+  queries: DataQuery[];
+  datasourceUid?: string;
```

Comment:
why does this need `queries` or `datasourceUid`? 🤔

Reply:
The button will actually implement anything that happens with the query when you click "add query" or what ever the action is going to be. For example in explore it basically goes into the explore redux state to do the necessary changes. So for doing anything useful you need the query that was clicked on, and it's data source (because query object don't contain the data source, only if the data source is mixed as far as I know)